lib/tasks/blazer.rake

```diff
@@ -0,0 +1,154 @@
+
+namespace :blazer do
+  desc 'Add Blazer db security for all organizations'
+  task add_database_security: :environment do
+    check_blazer_config
+
+    # read-only users per organization
+    Organization.all.each do |org|
+      run_query create_organization_user(org)
+    end
+
+    # policies to restrict organizations to their own data
+    db_connection.tables.each do |table|
+      if db_connection.columns(table).map(&:name).include?("organization_id")
+        run_query policy_sql(table)
+      end
+    end
+  end
+
+  desc 'Remove Blazer db security for organizations'
+  task drop_database_security: :environment do
+    check_blazer_config
+
+    # policies to restrict organizations to their own data
+    db_connection.tables.each do |table|
+      if db_connection.columns(table).map(&:name).include?("organization_id")
+        run_query drop_policy_sql(table)
+      end
+    end
+
+    # read-only users per organization
+    Organization.all.each do |org|
+      run_query drop_organization_user(org)
+    end
+  end
+
+private
+
+  def saving?
+    ENV['SAVE']
+  end
+
+  # Does blazer.yml contain a user for all organizations?
+  # We can't create a postgres user otherwise.
+  def check_blazer_config
+    orgs = Organization.ids.map { |id| "org#{id}".to_sym }
+    blazer_orgs = blazer_config[:data_sources].keys - [:main]
+
+    missing_orgs = orgs - blazer_orgs
+    return if missing_orgs.empty?
+
+    puts <<~MESSAGE
+      [ERROR] Missing user#{'s' if missing_orgs.size > 1} from config/blazer.yml file:
+      -> #{missing_orgs.join("\n- ")}
+    MESSAGE
+    exit
+  end
+
+  # This follows the recommended SQL from [blazer's docs](https://github.com/ankane/blazer#postgresql)
+  def create_organization_user(org)
+    org_user = "org#{org.id}"
+
+    <<~SQL
+      BEGIN;
+      CREATE ROLE #{org_user} LOGIN PASSWORD '#{org_db_password(org)}';
+      GRANT CONNECT ON DATABASE #{db_connection.current_database} TO #{org_user};
+      GRANT USAGE ON SCHEMA public TO #{org_user};
+      GRANT SELECT ON ALL TABLES IN SCHEMA public TO #{org_user};
+      ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT SELECT ON TABLES TO #{org_user}; 
+      COMMIT;
+    SQL
+  end
+
+  def drop_organization_user(org)
+    org_user = "org#{org.id}"
+
+    <<~SQL
+      DROP OWNED BY #{org_user};
+      DROP ROLE #{org_user};
+    SQL
+  end
+
+  def policy_sql(table)
+    <<~SQL
+      ALTER TABLE #{table} ENABLE ROW LEVEL SECURITY;
+      CREATE POLICY org_policy ON #{table}
+      FOR SELECT
+      TO public
+      USING (CASE #{policy_case_conditions}
+             END);
+    SQL
+  end
+
+  def drop_policy_sql(table)
+    <<~SQL
+      ALTER TABLE #{table} DISABLE ROW LEVEL SECURITY;
+      DROP POLICY org_policy ON #{table};
+    SQL
+  end
+
+  def policy_case_conditions
+    Organization.ids.map do |id|
+      "WHEN CURRENT_USER = 'org#{id}' THEN organization_id = #{id}"
+    end.join("\n            ")
+  end
```

Comment:
Nice! I think this is a huge first step and POC not just for blazer but for organizational security throughout the app.